library/nsxt_global_manager_active.py

```diff
@@ -0,0 +1,172 @@
+    for global_manager in global_managers['results']:
+        if global_manager.__contains__('display_name') and global_manager['display_name'] == display_name:
+            return global_manager
+    return None
+
+
+def wait_till_switchover_complete(module, url, mgr_username, mgr_password, validate_certs):
+    try:
+        retry_count = 0
+        while True:
+            (rc, resp) = request(url, headers=dict(Accept='application/json'),
+                                 url_username=mgr_username, url_password=mgr_password, validate_certs=validate_certs,
+                                 ignore_errors=True)
+            if resp['overall_status'] == "ONGOING" and retry_count < 100:
+                time.sleep(10)
+            elif resp['overall_status'] == "COMPLETE":
+                return
+            else:
+                all_erroes = ''
+                if resp['errors'] is not None:
+                    for e in resp['errors']:
+                        all_erroes = all_erroes + e
+                module.fail_json(msg='Switchover was not completed due to errors : %s' % all_erroes)
+    except Exception as err:
+        module.fail_json(msg='Error checking switchover status. Error [%s]' % (to_native(err)))
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(display_name=dict(required=True, type='str'))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    mgr_hostname = module.params['hostname']
+    mgr_username = module.params['username']
+    mgr_password = module.params['password']
+    validate_certs = module.params['validate_certs']
+    display_name = module.params['display_name']
+
+    manager_url = 'https://{}/global-manager/api/v1'.format(mgr_hostname)
+    global_manager_url = manager_url + '/global-infra/global-managers'
+    switchover_api_url = 'https://{}/api/v1/sites/switchover-status'.format(mgr_hostname)
+
+    existing_global_manager = get_global_manager_from_display_name(module, global_manager_url,
+                                                                   mgr_username, mgr_password, validate_certs,
+                                                                   display_name)
+    global_manager_id, revision = None, None
+
+    if existing_global_manager is None:
+        module.fail_json(msg="Global_manager with disaplya name [%s] not found." % display_name)
+
+    global_manager_id = existing_global_manager['id']
+    revision = existing_global_manager['_revision']
+
+    if existing_global_manager["mode"] == "ACTIVE":
+        module.exit_json(changed=False, id=global_manager_id,
+                         message="Global manager with display name %s is already in ACTIVE mode." %
+                                 existing_global_manager["display_name"])
+    else:
+        headers = dict(Accept="application/json")
+        headers['Content-Type'] = 'application/json'
+        request_data_dict = {
```

Comment:
It would be much better if this dictionary is returned as a call to a function..

Reply:
This API is only to change status from STANBY to ACTIVE. Its less likely to grow like other POST APIs.